src/Pact/Native.hs

```diff
@@ -206,31 +319,23 @@ langDefs =
     ,defRNative "identity" identity (funType a [("value",a)])
      "Return provided value. `(map (identity) [1 2 3])`"
 
-    ,defRNative "hash" hash' (funType tTyString [("value",a)])
-     "Compute BLAKE2b 512-bit hash of VALUE. Strings are converted directly while other values are \
-     \converted using their JSON representation. `(hash \"hello\")` `(hash { 'foo: 1 })`"
+    ,hashDef
 
     ])
-    where a = mkTyVar "a" []
-          b = mkTyVar "b" []
+    where b = mkTyVar "b" []
           c = mkTyVar "c" []
           d = mkTyVar "d" []
           row = mkSchemaVar "row"
           yieldv = TySchema TyObject (mkSchemaVar "y")
           obj = tTyObject (mkSchemaVar "o")
-          listA = mkTyVar "a" [TyList (mkTyVar "l" []),TyPrim TyString,TySchema TyObject (mkSchemaVar "o")]
           listStringA = mkTyVar "a" [TyList (mkTyVar "l" []),TyPrim TyString]
           takeDrop = funType listStringA [("count",tTyInteger),("list",listStringA)] <>
                      funType obj [("keys",TyList tTyString),("object",obj)]
           lam x y = TyFun $ funType' y [("x",x)]
           lam2 x y z = TyFun $ funType' z [("x",x),("y",y)]
 
-
-if' :: NativeFun e
-if' i as@[cond,then',else'] = gasUnreduced i as $ reduce cond >>= \cm -> case cm of
-           TLiteral (LBool c) _ -> reduce (if c then then' else else')
-           t -> evalError' i $ "if: conditional not boolean: " ++ show t
-if' i as = argsError' i as
+a :: Type n
```

Comment:
Hmmm.... I just noticed this coming from "why did `a` change to `a'` below. Not the greatest function name now that it's top-level. 

Fine to leave it for now, but put this on the list of "now that we know what native defs look like" (ie, with this nice formatting improvement) we might want to consider a quasiquoter or something to make writing the native defs nice so that they "read like" pact, removing the need for these type-signature masquerade functions.